week2/The system of the student's score management. v1.0.cpp

```diff
@@ -0,0 +1,160 @@
+#include <stdio.h>
+
+void readscore(int n, long student_number[], int score[]);
+void print_total_and_average(int n, int score[]);
+void rank_by_grade_in_descending_order(int n, long student_number[], int score[]);
+void print_score(int n, long student_number[], int score[]);
+void rank_by_student_number_in_ascending_order(int n, long student_number[], int score[]);
+int search_student_rank_and_grade_by_student_number(int n, long student_number[], int score[]);
+void count_the_percentage_of_the_each_category(int n, long student_number[], int score[]);
+
+int main()
+{
+	long student_number[30];
+	int score[30], n, _switch;
+	printf("1.Input record\n2.Caculate total and average score of course\n3.Sort in descending order by score\n");
+	printf("4.Sort in ascending order by number\n5.search by number\n6.Statistic analysis\n0.Exit\n\n");
+	do {
+		printf("Please enter your choice:");
+		scanf_s("%d", &_switch);
+		switch (_switch)
+		{
+			case 1:
+				printf("Please input the number of the students:");
+				scanf_s("%d", &n);
+				readscore(n, student_number, score);
+				break;
+			case 2:
+				print_total_and_average(n, score);
+				break;
+			case 3:
+				rank_by_grade_in_descending_order(n, student_number, score);

```

Comment:
This is good because we just want to use an in-place sort here.

Just want to share a programming principle. It would be better not to change the order of the original array in a function because this will cause some unexpected result - this scenario is so-called "side effect". In this case, consider if we have another option in this program, which is just to print the score array. Its result will be unstable before and after different sorting functions' execution.

A common practice to avoid side effect is, instead of using a `void` function, we will return a new array that consists of the sorting result.

Reply:
How to do it?

Reply:
In the function body, you can make a copy of original array, then sort the copied array and return it back.

Reply:
OK